src/three/TilesRenderer.js

```diff
@@ -647,6 +589,18 @@ export class TilesRenderer extends TilesRendererBase {
 
 			}
 
+
+			case 'subt': {
+
+				const loader = new SUBTREELoader( manager, tile, this.root );
+				loader.workingPath = workingPath;
+				loader.fetchOptions = fetchOptions;
+				loader.parse( buffer );
+				promise = Promise.resolve( null );
+				break;
+
+			}
```

Comment:
Any subtree parsing should happen in the `TilesRendererBase` class since there's nothing three.js specific about it. The goal in the TilesRenderer and TilesRendererBase separation was to keep any no-three.js, 3d-tiles-specific logic in the "Base" class so it's easier to follow. See how external tile sets are loaded in [requestTileConents](https://github.com/NASA-AMMOS/3DTilesRendererJS/blob/master/src/base/TilesRendererBase.js#L611).

Reply:
Do you mean that I should do 
```
if(isExternalTileSet){
...
} else if(tile.isSubtree){

SAME CODE AS THE CASE !isExternalTileSet
Except I call : 
const loader = new SUBTREELoader( manager, tile, this.root );
loader.workingPath = workingPath;
loader.fetchOptions = fetchOptions;
loader.parse( buffer );

Instead of parseTile

}else{
... 
}

```
Is it correct ?
Could you draft a piece of script so its easier for me to understand ? 

Thank you

Reply:
Yeah let's start with that, for now, and we can dedupe some of the load logic later on. Here's an example of what I'm imagining with some abbreviated function contents. Here "SubTreeLoader" is returning a parsed root with children that is shaped like any other 3d tiles tile from the schema:

```js
if ( isExternalTileSet ) {

    // ...

} else if ( /subt$/i.test( tile.content.uri ) ) {

    downloadQueue.add( tile, tileCb => {

        // ...

        return fetch( processedUrl, Object.assign( { signal }, this.fetchOptions ) );

    } )
        .then( buffer => {

            // ...

            return res.arrayBuffer();

        } )
        .then( buffer => {

            // ...

            // handle parsing of subtree data and assignment of tile children
            const loader = new SubTreeLoader();
            const subTreeRoot = loader.parse( buffer );
            tile.children.push( subTreeRoot );

        } )
        .catch( errorCallback );

} else {

    // ...

}
```

Let me know if that answers things